### PR TITLE
[NETBEANS-4378] Ensure module info in catalog and module info is aligned

### PR DESCRIPTION
**Pull request by matthiasblaesing**
The algorithm in PP3 generated codename base and target cluster based
on the maven coordinates. These are not aligned and thus the catalog
contained information disconnected from the actual module.

At installation time this caused the IDE to not finding the modules.

### CODE REVIEW

**Comment by matthiasblaesing** [comment]
@jpirek could you please have a look at this?

For the background please have a look at the issue report: https://issues.apache.org/jira/browse/NETBEANS-4378

This basicly switches from using the maven coordinates to using the info.xml embedded into the module.

**Comment by jkovalsky** [comment]
@matthiasblaesing I assume you have tested it locally, haven't you?

**Comment by matthiasblaesing** [comment]
There is a test instance here (only one plugin, but I verified, that the problem was reproducible before):

https://doppel-helix.eu/pp3/

I wanted to get this out so that more eyes can get onto this. 

One thing I forgot yesterday was that JAR plugins were mentioned in the code. Are JARs really supported and how are they installed?

**Comment by ebarboni** [comment]
@matthiasblaesing @jkovalsky once this fixed can you warn me so I can proceed to release preparation. (If no impact in IDE code of course)

**Comment by matthiasblaesing** [comment]
I knew I should not have tested, but well I think I got the code into a state where NB Autoupdate code will accept plugins from the center.

I pushed three new commits into this branch - I'd like a second look. For details please have a look at the individual commits I added comments, that try to explain the updates.

**Comment by matthiasblaesing** [comment]
The testing of the OSGI jar support is a bit more involved, than NBM testing. The OSGI jars are not shown in the update center and are only pulled in as dependencies. Thus I modified the contents of the info.xml for the LDIF explorer to depend on rhino:

https://doppel-helix.eu/pp3/data/11.2/catalog.xml

```xml
<manifest OpenIDE-Module="com.google.code.nb_ldap_explorer.ldif" 
          OpenIDE-Module-Name="LDIF Editor" 
          OpenIDE-Module-Specification-Version="0.15" 
          OpenIDE-Module-Implementation-Version="0.15"
          OpenIDE-Module-Module-Dependencies="org.openide.filesystems &gt; 9.12.1, org.openide.loaders &gt; 7.68.1, org.openide.nodes &gt; 7.47.1, org.netbeans.modules.editor/3 &gt; 1.89.1.6.3.23.55, org.netbeans.modules.parsing.api/1 &gt; 9.9.1.8, org.openide.text &gt; 6.69.1, org.netbeans.modules.lexer/2 &gt; 1.65.1.1, org.openide.util &gt; 9.9.1, org.openide.util.lookup &gt; 8.35.1, org.mozilla.rhino" 
          OpenIDE-Module-Short-Description="Syntax Highlighting for LDIF-Files (intended as a companion to the ldap explorer)" 
          OpenIDE-Module-Long-Description="Syntax Highlighting for LDIF-Files (intended as a companion to the ldap explorer)"
          OpenIDE-Module-Display-Category="Tools"
          OpenIDE-Module-Requires="org.openide.modules.ModuleFormat1"/>
```

This change was done in the DB, so if anyone tries to replicate, this is required. With this change, installing the LDIF Editor will cause rhino to be pulled in additionally. Easiest was to verify:

- installation must work and not timeout
- Installing an OSGI bundle through the plugin portal will create a new cluster pluginportal, where the osgi bundles will be placed.

**Comment by matthiasblaesing** [comment]
@jkovalsky thanks for checking!

File: pp3/module/Application/src/Application/Pp/Catalog.php [hunk 1]
```diff
@@ -120,7 +205,89 @@ public function storeXml($destinationFolder, $xml) {
...
+               exec($execution, $output, $return_var);
+               $outputString = implode("\n", $output);
+               if(intval($return_var) != 0) {
+                   error_log(sprintf('PluginVersion(id: %d) Failed to extract info from JAR, Got: %s', $pluginVersion->getId(), $outputString));
+                   return;
+               }
+               $infoXML = $outputString;
+            } else {
+                // Extract Info/info.xml from archive
+                $archive = new ZipArchive();
```
Comment (jkovalsky) [inline pp3/module/Application/src/Application/Pp/Catalog.php:275]: BTW, we are seeing this error in the server log:
`Error: Class 'ZipArchive' not found in /space/src/netbeans-tools/pp3/module/Application/src/Application/Pp/Catalog.php:275`

Comment (matthiasblaesing) [reply pp3/module/Application/src/Application/Pp/Catalog.php:275]: On which server and is this recent? It was observed in production prior to the installation of php7.0-zip, after that the required class is present and can be loaded. As the regeneration after the update worked, the archives must have been readable.

Comment (jkovalsky) [reply pp3/module/Application/src/Application/Pp/Catalog.php:275]: Yes, apologies. I am terribly behind with e-mails so I didn't know you have already fixed it over the weekend whereas that error was from Friday. Great job Matthias/Eric/Antonio!

**Pull request merged**